Mathlib/RingTheory/RootsOfUnity/More.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2024 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.FieldTheory.KummerExtension
+import Mathlib.RingTheory.RootsOfUnity.Basic
+
+/-!
+# More results on primitive roots of unity
+
+(We put these in a separate file because of the `KummerExtension` import.)
+
+Assume that `μ` is a primitive `n`th root of unity in an integral domain `R`. Then
+$$ \prod_{k=1}^{n-1} (1 - \mu^k) = n \,; $$
+see `IsPrimitiveRoot.prod_one_sub_pow_eq_order` and its variant
+`IsPrimitiveRoot.prod_pow_sub_one_eq_order` in terms of `∏ (μ^k - 1)`.
+
+We use this to deduce that `n` is divisible by `(μ - 1)^k` in `ℤ[μ] ⊆ R` when `k < n`.
+-/
+
+/-- If `A` is an `R`-algebra and `μ : A`, then `μ^k-1` is divisible by `μ-1` in `R[μ] ⊆ A`. -/
+lemma Algebra.adjoin.sub_one_dvd_pow_sub_one (R : Type*) {A : Type*} [CommRing R] [CommRing A]
```

Comment:
Can you use `∣` explicitly here?

Reply:
I was also thinking of inlining this in the proof of `order_eq_mul_self_sub_one_pow`, as it is a fairly specialized statement (with a fairly short proof) whose usefulness outside `order_eq_mul_self_sub_one_pow` is perhaps questionable.

Reply:
I have now inlined `Algebra.adjoin.sub_one_dvd_pow_sub_one`.